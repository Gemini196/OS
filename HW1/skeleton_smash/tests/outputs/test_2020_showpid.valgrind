==68774== Memcheck, a memory error detector
==68774== Copyright (C) 2002-2017, and GNU GPL'd, by Julian Seward et al.
==68774== Using Valgrind-3.13.0 and LibVEX; rerun with -h for copyright info
==68774== Command: /home/student/Desktop/OS/HW1/skeleton_smash/smash
==68774== Parent PID: 68770
==68774== 
==68774== Invalid read of size 1
==68774==    at 0x5488B30: __strcmp_sse2_unaligned (strcmp-sse2-unaligned.S:24)
==68774==    by 0x10DA45: QuitCommand::execute() (Commands.cpp:627)
==68774==    by 0x10F120: SmallShell::executeCommand(char const*) (Commands.cpp:916)
==68774==    by 0x1109A1: main (smash.cpp:51)
==68774==  Address 0x0 is not stack'd, malloc'd or (recently) free'd
==68774== 
==68774== 
==68774== Process terminating with default action of signal 11 (SIGSEGV)
==68774==  Access not within mapped region at address 0x0
==68774==    at 0x5488B30: __strcmp_sse2_unaligned (strcmp-sse2-unaligned.S:24)
==68774==    by 0x10DA45: QuitCommand::execute() (Commands.cpp:627)
==68774==    by 0x10F120: SmallShell::executeCommand(char const*) (Commands.cpp:916)
==68774==    by 0x1109A1: main (smash.cpp:51)
==68774==  If you believe this happened as a result of a stack
==68774==  overflow in your program's main thread (unlikely but
==68774==  possible), you can try to increase the size of the
==68774==  main thread stack using the --main-stacksize= flag.
==68774==  The main thread stack size used in this run was 8388608.
==68774== 
==68774== FILE DESCRIPTORS: 6 open at exit.
==68774== Open file descriptor 1: /home/student/Desktop/OS/HW1/skeleton_smash/tests/outputs/test_2020_showpid.out
==68774==    at 0x54EF957: dup2 (syscall-template.S:78)
==68774==    by 0x1108F8: main (smash.cpp:41)
==68774== 
==68774== Open file descriptor 0:
==68774==    at 0x54EF957: dup2 (syscall-template.S:78)
==68774==    by 0x1108A6: main (smash.cpp:33)
==68774== 
==68774== Open file descriptor 5:
==68774==    at 0x54EF927: dup (syscall-template.S:78)
==68774==    by 0x110A67: __static_initialization_and_destruction_0(int, int) (smash.cpp:9)
==68774==    by 0x110A83: _GLOBAL__sub_I_stdout_dp (smash.cpp:57)
==68774==    by 0x110AFC: __libc_csu_init (in /home/student/Desktop/OS/HW1/skeleton_smash/smash)
==68774==    by 0x5400C17: (below main) (libc-start.c:266)
==68774== 
==68774== Open file descriptor 4: /home/student/Desktop/OS/HW1/skeleton_smash/tests/outputs/test_2020_showpid.out
==68774==    at 0x54EF927: dup (syscall-template.S:78)
==68774==    by 0x110A57: __static_initialization_and_destruction_0(int, int) (smash.cpp:8)
==68774==    by 0x110A83: _GLOBAL__sub_I_stdout_dp (smash.cpp:57)
==68774==    by 0x110AFC: __libc_csu_init (in /home/student/Desktop/OS/HW1/skeleton_smash/smash)
==68774==    by 0x5400C17: (below main) (libc-start.c:266)
==68774== 
==68774== Open file descriptor 3: /home/student/Desktop/OS/HW1/skeleton_smash/tests/outputs/test_2020_showpid.valgrind
==68774==    <inherited from parent>
==68774== 
==68774== Open file descriptor 2: /home/student/Desktop/OS/HW1/skeleton_smash/tests/outputs/test_2020_showpid.err
==68774==    <inherited from parent>
==68774== 
==68774== 
==68774== HEAP SUMMARY:
==68774==     in use at exit: 669 bytes in 9 blocks
==68774==   total heap usage: 52 allocs, 43 frees, 82,764 bytes allocated
==68774== 
==68774== 5 bytes in 1 blocks are still reachable in loss record 1 of 9
==68774==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==68774==    by 0x10B299: _parseCommandLine(char const*, char**) (Commands.cpp:47)
==68774==    by 0x10DA28: QuitCommand::execute() (Commands.cpp:625)
==68774==    by 0x10F120: SmallShell::executeCommand(char const*) (Commands.cpp:916)
==68774==    by 0x1109A1: main (smash.cpp:51)
==68774== 
==68774== 16 bytes in 1 blocks are still reachable in loss record 2 of 9
==68774==    at 0x4C3217F: operator new(unsigned long) (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==68774==    by 0x110845: main (smash.cpp:24)
==68774== 
==68774== 24 bytes in 1 blocks are still reachable in loss record 3 of 9
==68774==    at 0x4C3217F: operator new(unsigned long) (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==68774==    by 0x10E0BD: JobsList::JobsList() (Commands.cpp:707)
==68774==    by 0x110850: main (smash.cpp:24)
==68774== 
==68774== 24 bytes in 1 blocks are still reachable in loss record 4 of 9
==68774==    at 0x4C3217F: operator new(unsigned long) (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==68774==    by 0x10EF5E: SmallShell::CreateCommand(char const*) (Commands.cpp:897)
==68774==    by 0x10F07A: SmallShell::executeCommand(char const*) (Commands.cpp:904)
==68774==    by 0x1109A1: main (smash.cpp:51)
==68774== 
==68774== 24 bytes in 1 blocks are indirectly lost in loss record 5 of 9
==68774==    at 0x4C3217F: operator new(unsigned long) (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==68774==    by 0x10E0BD: JobsList::JobsList() (Commands.cpp:707)
==68774==    by 0x10EB3C: SmallShell::SmallShell() (Commands.cpp:860)
==68774==    by 0x110752: SmallShell::getInstance() (Commands.h:233)
==68774==    by 0x110824: main (smash.cpp:21)
==68774== 
==68774== 40 (16 direct, 24 indirect) bytes in 1 blocks are definitely lost in loss record 6 of 9
==68774==    at 0x4C3217F: operator new(unsigned long) (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==68774==    by 0x10EB31: SmallShell::SmallShell() (Commands.cpp:860)
==68774==    by 0x110752: SmallShell::getInstance() (Commands.h:233)
==68774==    by 0x110824: main (smash.cpp:21)
==68774== 
==68774== 160 bytes in 1 blocks are still reachable in loss record 7 of 9
==68774==    at 0x4C3289F: operator new[](unsigned long) (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==68774==    by 0x10D9E0: QuitCommand::execute() (Commands.cpp:620)
==68774==    by 0x10F120: SmallShell::executeCommand(char const*) (Commands.cpp:916)
==68774==    by 0x1109A1: main (smash.cpp:51)
==68774== 
==68774== 200 bytes in 1 blocks are still reachable in loss record 8 of 9
==68774==    at 0x4C3289F: operator new[](unsigned long) (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==68774==    by 0x10EB4E: SmallShell::SmallShell() (Commands.cpp:861)
==68774==    by 0x110752: SmallShell::getInstance() (Commands.h:233)
==68774==    by 0x110824: main (smash.cpp:21)
==68774== 
==68774== 200 bytes in 1 blocks are still reachable in loss record 9 of 9
==68774==    at 0x4C3289F: operator new[](unsigned long) (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==68774==    by 0x10D9EE: QuitCommand::execute() (Commands.cpp:621)
==68774==    by 0x10F120: SmallShell::executeCommand(char const*) (Commands.cpp:916)
==68774==    by 0x1109A1: main (smash.cpp:51)
==68774== 
==68774== LEAK SUMMARY:
==68774==    definitely lost: 16 bytes in 1 blocks
==68774==    indirectly lost: 24 bytes in 1 blocks
==68774==      possibly lost: 0 bytes in 0 blocks
==68774==    still reachable: 629 bytes in 7 blocks
==68774==         suppressed: 0 bytes in 0 blocks
==68774== 
==68774== For counts of detected and suppressed errors, rerun with: -v
==68774== ERROR SUMMARY: 2 errors from 2 contexts (suppressed: 0 from 0)
