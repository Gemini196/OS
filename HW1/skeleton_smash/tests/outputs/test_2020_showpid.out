smash> smash pid is 46759
smash> smash pid is 46759
smash> smash> smash pid is 46759
smash> smash pid is 46759
smash> smash pid is 46759
smash> smash> 